packages/teleport/src/Discover/DownloadScript/useDownloadScript.tsx

```diff
@@ -24,13 +29,18 @@ import { AgentStepProps } from '../types';
 
 import type { JoinToken } from 'teleport/services/joinToken';
 
-const FIVE_MINUTES_IN_MS = 300000;
+const FIVE_MINUTES = 5;
 const THREE_SECONDS_IN_MS = 3000;
+const ONE_SECOND_IN_MS = 1000;
 
 export function useDownloadScript({ ctx, props }: Props) {
   const { attempt, run, setAttempt } = useAttempt('processing');
   const [joinToken, setJoinToken] = useState<JoinToken>();
   const [pollState, setPollState] = useState<PollState>('polling');
+  const [countdownTime, setCountdownTime] = useState<CountdownTime>({
```

Comment:
The logic for this countdown is spread out all over the place. I think it would be a lot easier to maintain (and use elsewhere if needed) if it was its own separate component

`<Countdown time="00:05:00" />`

Reply:
if there is no faulty logic, i would like to keep this as is, until it's time to refactor (ill put a TODO marker)